examples/android-client/app/src/main/java/io/dronecore/dronecoreclient/MapsActivity.java

```diff
@@ -0,0 +1,214 @@
+import androidx.appcompat.app.AppCompatActivity;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.BitmapDescriptorFactory;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.dronecode_sdk.action.Action;
+import io.dronecode_sdk.mission.Mission;
+import io.dronecode_sdk.telemetry.Telemetry;
+import io.reactivex.disposables.Disposable;
+
+import static com.google.android.gms.maps.GoogleMap.MAP_TYPE_HYBRID;
+
+/**
+ * Main Activity to display map and create missions.
+ * 1. Take off
+ * 2. Long click on map to add a waypoint
+ * 3. Touch an existing waypoint to delete it
+ * 4. Hit play to start mission.
+ */
+public class MapsActivity extends AppCompatActivity implements OnMapReadyCallback, GoogleMap.OnMarkerDragListener {
+
+    public static final String TAG = "MapsActivity";
+    public static final String BACKEND_IP_ADDRESS = "10.42.0.65"; // todo update this to ip address of device running the backend. remove this once the backend is included in the app
+
+    private GoogleMap map = null;
+    private MapsViewModel viewModel;
+    private Marker currentPositionMarker;
+    private final List<Marker> markers = new ArrayList<>();
+    private Action action;
+    private Mission mission;
+    private Telemetry telemetry;
+    private final List<Disposable> disposables = new ArrayList<>();
+
+    private Observer<LatLng> currentPositionObserver = this::updateVehiclePosition;
+    private Observer<List<LatLng>> currentMissionPlanObserver = this::updateMarkers;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_maps);
+        viewModel = ViewModelProviders.of(this).get(MapsViewModel.class);
+
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager().findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this);
+
+        FloatingActionButton floatingActionButton = findViewById(R.id.fab);
+        floatingActionButton.setOnClickListener(v -> viewModel.startMission(mission));
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        viewModel.currentPositionLiveData.observe(this, currentPositionObserver);
+        viewModel.currentMissionPlanLiveData.observe(this, currentMissionPlanObserver);
+        action = new Action(BACKEND_IP_ADDRESS, 50051); // TODO: 4/4/19 50051 should be a constant in the sdk somewhere
```

Comment:
Good point for `50051`. Actually in other language wrappers (python, swift) we have a `Drone` object that instantiates the plugins for you, so you do `Drone drone = new Drone();` and then you can use `drone.action.arm()` etc.